src/tests/resources/sparse_matrix_environment.cpp

```diff
@@ -0,0 +1,37 @@
+/* ************************************************************************
+* Copyright 2015 Advanced Micro Devices, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+* ************************************************************************ */
+#include "sparse_matrix_environment.h"
+
+CSRSparseEnvironment::sMatrixType CSRSparseEnvironment::ublasSCsr = CSRSparseEnvironment::sMatrixType();
+
+#ifdef TEST_LONG
```

Comment:
This was to test matrices with very long rows/columns compared to other dimension. rows >> cols or cols >> rows.